doc/auto_storage.md

```diff
@@ -0,0 +1,663 @@
+  property <string>
+  value <any>
+  operator [<Operator='equal'>]
+
+Operator <'equal'|'notEqual'|'less'|'greater'|'lessOrEqual'|'greaterOrEqual'>
+
+Sort
+  property <string>
+  order <'asc'|'desc'>
+
+NotFoundAction <'continue'|'skip'|'error'>
+```
+
+For example, `search` sections could be used to find the three biggest disks in the system, delete
+all linux partitions bigger than 1 GiB within them and create new partitions of type RAID.
+
+```json
+"storage": {
+    "drives": [
+        {
+            "search": {
+                "sort": { "property": "sizeKib", "order": "desc" },
+                "max": 3
+            },
+            "partitions": [
+                {
+                    "search": {
+                        "condition": {
+                            "and": [
+                                { "property": "id", "value": "linux" },
+                                { "property": "sizeGib", "value": 1, "operator": "bigger" }
+                            ]
+                        }
+                    },
+                    "delete": true
+                },
+                { 
+                    "alias": "newRaidPart",
+                    "id": "raid",
+                    "size": { "min": "1 GiB" }
+                }
+              ]
+        }
+    ]
+}
+```
+
+The example serves to illustrate several aspects of the searching mechanism.
+
+**FIXME:** The case of "drives" is special because they can only be matched to existing devices,
+not created. That makes searching different from other kind of devices at several levels. For
+example, the reasonable default values for `min` and `max` or the exact meaning of `ifNotFound`
+(which could be a way to implement find-or-create) would be affected.
+
+First of all, the scope of each search. That is, the devices from the system that are considered as
+possible candidates. That obviously depends on the place in the profile of the `search` section.
+A `search` section inside the description of an MD RAID will only match MD devices and a `search`
+section inside the `partitions` subsection of that RAID description will only match partitions of
+RAIDs that have matched the conditions of the most external `search`.
+
+Another aspect that may need some clarification is the usage of `min` and `ifNotFound`. The former
+can be combined with `max` to specify how many real devices are expected to match the definition and
+a value of 0 implies that is perfectly fine if no device is found. The latter defines what to do
+with the section if the minimum amount of devices is not reached...
+
+**FIXME:** how to handle all scenarios. For example, if defining volume groups with a min of 3
+but one is found? Should we create two more to reach the min or process the one we found? There are
+many combinations to discuss.
+
+**FIXME:** crazy idea: having min and max out of `search` (that is, also for new, non-searched,
+devices) in order to create several VGs or partitions or whatever at once.
```

Comment:
I think this could complicate the profile. For example, what would `min` and `max` mean for a VG without `search`? How do we indicate the PVs for each VG to create?

Reply:
I removed the mention to the crazy idea. It was just that.